lib/ansible/modules/windows/win_robocopy.ps1

```diff
@@ -92,74 +93,61 @@ $result.flags = $flags
 
 $robocopy_output = ""
 $rc = 0
-If ($check_mode -eq $true) {
-    $robocopy_output = "Would have copied the contents of $src to $dest"
-    $rc = 0
-}
-Else {
-    Try {
-        &robocopy $robocopy_opts | Tee-Object -Variable robocopy_output | Out-Null
-        $rc = $LASTEXITCODE
-    }
-    Catch {
-        $ErrorMessage = $_.Exception.Message
-        Fail-Json $result "Error synchronizing $src to $dest! Msg: $ErrorMessage"
-    }
+Try {
+
+    &robocopy $robocopy_opts | Tee-Object -Variable robocopy_output | Out-Null
```

Comment:
I believe this is done this way due to some powershell quirks. If you run the process like
`&robocopy $robocopy_opts`
Powershell will run it and not wait until it is finished executing before continuing on. You can pipe it to something like Out-Null or Out-Host but if you want the output as a variable you need to use Tee-Object to redirect it to a variable. I believe the final Out-Null just stops it from displaying it on the host.